overlord/snapstate/snapstate.go

```diff
@@ -74,6 +74,35 @@ var ErrNothingToDo = errors.New("nothing to do")
 
 var osutilCheckFreeSpace = osutil.CheckFreeSpace
 
+type minimalInstallInfo interface {
```

Comment:
this interface seems fine, could you split out its introduction and the relative changes to preexisting code to make the PR a bit smaller?